airflow/contrib/hooks/docker_hook.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from docker import Client
+from docker.errors import APIError
+
+from airflow.exceptions import AirflowException
+from airflow.hooks.base_hook import BaseHook
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+class DockerHook(BaseHook, LoggingMixin):
+    """
+    Interact with a private Docker registry.
+
+    :param docker_conn_id: ID of the Airflow connection where
+        credentials and extra configuration are stored
+    :type docker_conn_id: str
+    """
+    def __init__(self,
+            docker_conn_id='docker_default',
+            base_url='unix://var/run/docker.sock',
+            version='auto',
+            tls=None
+            ):
+        self.__docker_conn_id = docker_conn_id
+        self.__base_url = base_url
+        self.__version = version
+        self.__tls = tls
+        self.client = None
+
+    def get_conn(self):
+        self.log.debug('Creating Docker client for conn_id: %s', self.__docker_conn_id)
+        conn = self.get_connection(self.__docker_conn_id)
+        self.client = Client(
+            base_url=self.__base_url,
+            version=self.__version,
+            tls=self.__tls
+        )
+        self.__login()
+        return self.client
+
+    def __login(self):
+        self.log.debug('Logging into Docker registry')
+        conn = self.get_connection(self.__docker_conn_id)
+        extra_options = conn.extra_dejson
+        if not conn.host:
```

Comment:
Why don't you check this in init?

Reply:
I thought that `get_conn()` was responsible for this. My blueprint was [the SSH hook](https://github.com/moertel/incubator-airflow/blob/99954b245321b58b22a8907e5f71f3cd71174942/airflow/contrib/hooks/ssh_hook.py#L96) which does it the same way.